Sources/PopupBridge/POPPopupBridge.swift

```diff
@@ -1,63 +1,61 @@
-public class POPPopupBridge: NSObject, WKScriptMessageHandler, SFSafariViewControllerDelegate {
+public class POPPopupBridge: NSObject, WKScriptMessageHandler {
+
+    /// Exposed for testing
+    var returnedWithURL: Bool = false
     
     // MARK: - Private Properties
     
     private let messageHandlerName = "POPPopupBridge"
-    private let hostName = "popupbridgev1"
-    
+    private let hostName = "popupbridgev1"    
     private let webView: WKWebView
-    private let urlScheme: String
-    private let delegate: POPPopupBridgeDelegate
-    private var safariViewController: SFSafariViewController?
+    private var webAuthenticationSession: WebAuthenticationSession
     
-    private static var returnBlock: ((URL) -> Bool)?
+    private var returnBlock: ((URL) -> Void)? = nil
     
-    // MARK: - Initializer
+    // MARK: - Initializers
         
     /// Initialize a Popup Bridge.
     /// - Parameters:
     ///   - webView: The web view to add a script message handler to. Do not change the web view's configuration or user content controller after initializing Popup Bridge.
-    ///   - urlScheme: The URL Scheme that you have registered in your Info.plist.
-    ///   - delegate: A delegate that presents and dismisses the pop-up (a SFSafariViewController).
-    public init(
-        webView: WKWebView,
-        urlScheme: String,
-        delegate: POPPopupBridgeDelegate
-    ) {
+    public init(webView: WKWebView) {
         self.webView = webView
-        self.urlScheme = urlScheme
-        self.delegate = delegate
+        self.webAuthenticationSession = WebAuthenticationSession()
         
         super.init()
-        
-        configureWebView(scheme: urlScheme)
-        
-        POPPopupBridge.returnBlock = { (url : URL) -> Bool in
+
+        configureWebView(scheme: PopupBridgeConstants.callbackURLScheme)
+                
+        returnBlock = { url in
             guard let script = self.constructJavaScriptCompletionResult(returnURL: url) else {
-                return false
+                return
             }
-            
+
             self.injectWebView(webView: webView, withJavaScript: script)
-            return true
+            return
         }
     }
-    
-    // MARK: - Public Methods
-    
-    /// Handle completion of the popup flow by calling this method from either
-    /// your scene:openURLContexts: scene delegate method or
-    /// your application:openURL:sourceApplication:annotation: app delegate method.
-    @objc(openURL:)
-    public static func open(url: URL) -> Bool {
-        if let returnBlock {
-            return returnBlock(url)
-        } else {
-            return false
+
+    /// Exposed for testing
+    init(webView: WKWebView, webAuthenticationSession: WebAuthenticationSession) {
+        self.webView = webView
+        self.webAuthenticationSession = webAuthenticationSession
+
+        super.init()
+
+        configureWebView(scheme: PopupBridgeConstants.callbackURLScheme)
```

Comment:
The `configureWebView` func signature doesn't need to take a `scheme` anymore, it can just access `PopupBridgeConstants.callbackURLScheme` directly

Reply:
Updated: https://github.com/braintree/popup-bridge-ios/pull/57/commits/133f5611b1f883393b2d38a14ac4978b2cfabd09